src/master-playlist-controller.js

```diff
@@ -875,10 +880,11 @@ export class MasterPlaylistController extends videojs.EventTarget {
     const enabledPlaylists = playlists.filter(isEnabled);
     const isFinalRendition = enabledPlaylists.length === 1 && enabledPlaylists[0] === currentPlaylist;
 
-    if (playlists.length === 1) {
-      // Never blacklisting this playlist because it's the only playlist
-      videojs.log.warn('Problem encountered with the current ' +
-                       'playlist. Trying again since it is the only playlist.');
+    // Don't blacklist the only playlist unless it was blacklisted
+    // forever
+    if (playlists.length === 1 && blacklistDuration !== Infinity) {
```

Comment:
I don't think we should retry a playlist, even if it is the only one, when it is blacklisted with `Infinity`. Without this change, media with a single playlist cannot be stopped from looping downloads.